projects/core/src/occ/services/asm-context.service.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * SPDX-FileCopyrightText: 2022 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { HttpContext } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+import { OCC_ASM_TOKEN, OccAsmContext } from '../utils';
+
+/**
+ * Used to provide an ASM context for certain endpoints.
+ * For example, if an endpoint needs to pass in the emulated user's ID to resolve access issues,
+ * you should create a context.
+ * @see feature-libs/asm/root/interceptors/user-id.interceptor.ts
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class AsmContextService {
```

Comment:
For the sake of the discussion, would there be a way to make this context mechanism not ASM specific?  Ideally we'd avoid bringing back ASM logic back in the core lib.

Reply:
Hmm. I can rename the method `createASMContext` (same implementation) and thus rename the service to `HttpContextService`.

Reply:
Or completely remove it maybe? After all, it doesn’t do much and `HttpContext` is chainable. It’s the same as doing `new HttpContext().set(OCC_ASM_TOKEN, contextObj)` and the `OCC_ASM_TOKEN` is already exposed.